### PR TITLE
[aot] Add a depfile= argument which causes the AOT compiler to write out a gcc style dep file listing the dependencies between the output file and all referenced assemblies.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -7950,6 +7951,8 @@ mono_aot_parse_options (const char *aot_options, MonoAotOptions *opts)
 			opts->no_opt = TRUE;
 		} else if (str_begins_with (arg, "clangxx=")) {
 			opts->clangxx = g_strdup (arg + strlen ("clangxx="));
+		} else if (str_begins_with (arg, "depfile=")) {
```
Comment (marek-safar) [inline mono/mini/aot-compiler.c]: @vargaz could you please document this (including the expected format)

/cc @lambdageek

File: mono/mini/aot-compiler.c [hunk 2]
```diff
@@ -13504,6 +13508,32 @@ print_stats (MonoAotCompile *acfg)
...
+	g_assert (depfile);
+
+	int ntargets = 1;
+	char **targets = g_new0 (char*, ntargets);
+	targets [0] = acfg->aot_opts.llvm_outfile;
+	for (int tindex = 0; tindex < ntargets; ++tindex) {
+		fprintf (depfile, "%s: ", targets [tindex]);
+		for (int i = 0; i < acfg->image_table->len; i++) {
+			MonoImage *image = (MonoImage*)g_ptr_array_index (acfg->image_table, i);
+			fprintf (depfile, " %s", image->name);
```
Comment (lambdageek) [inline mono/mini/aot-compiler.c]: `image->filename`

File: man/mono.1 [hunk 3]
```diff
@@ -267,6 +267,9 @@ Print the architecture the AOT in this copy of Mono targets and quit.
 Generates all required wrappers, so that it is possible to run --interpreter without
 any code generation at runtime.  This option only makes sense with \fBmscorlib.dll\fR.
 Embedders can set
+.TP
+.I depfile=[filename]
+Outputs a gcc style dependency file.
```
Comment (lambdageek) [inline man/mono.1]: nit: `gcc -M` style

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**